specification/transport_layer/can.tex

```diff
@@ -317,16 +317,9 @@ \subsubsection{Tail byte}
 \subsubsection{Padding bytes}
 
 Certain transports (such as CAN FD) may not provide byte-level granularity of the CAN data field length.
-In that case, the useful payload is to be padded with the minimal number of padding bytes required
+In that case, the useful payload is to be padded with the minimal number of zero padding bytes required
 to bring the total length of the CAN data field to a value that can satisfy the length granularity constraints.
 
-When transmitting, each padding byte must be set to $\text{85} = \text{55}_\text{hex} = \text{0101\,0101}_\text{bin}$.
```

Comment:
padding is padding. Why do we have to change it to be zero padding? Edges matter for wireless protocols too so we're reducing the performance for both CAN and for any packet-oriented wireless networks we want to support.

Reply:
Padding is not always padding. The implicit (zero) extension rule makes deserialization routines sensitive to the trailing unused data. For example, a publisher emits an object of type:

```py
uint16 foo
```

Imagine that the transport layer requires padding to 4 bytes, which is done with 0x55. Suppose that the published value is 0x1234, so the resulting serialized representation is `[0x34, 0x12, 0x55, 0x55]`.

Suppose that the receiving side wants to rely on the implicit zero extension rule and use the following definition:

```py
uint16 foo
uint16 bar
```

The expectation is that `foo` will be deserialized as 0x1234, and `bar` will be zero-extended as 0x0000. If arbitrary padding is allowed, the value of `bar` becomes undefined; in this particular example it would be 0x5555.

Therefore, any implicit extension rule requires that padding is done with the same-valued bytes as the implicit extension. In our case it's zero.

An alternative would be to replace the zero-extension rule with 0x55-extension rule, but its utility for data type designers is far lower.

Reply:
Sorry, I should have been more clear: we can allow padding to be transmitted as non-zero as long as it is converted to zeros during deserialization...but then the data representation leaks into the driver...okay...and packet radios typically provide whitening capabilities (not unlike CAN's bit stuffing I suppose) so...never-mind. You're right.